internal/store/peering_connections.go

```diff
@@ -38,17 +38,17 @@ type PeeringConnectionCreator interface {
 }
 
 type AzurePeeringConnectionCreator interface {
-	AzureContainers(string) ([]atlasv2.AzureCloudProviderContainer, error)
+	AzureContainers(string) ([]*atlasv2.AzureCloudProviderContainer, error)
 	PeeringConnectionCreator
 }
 
 type AWSPeeringConnectionCreator interface {
-	AWSContainers(string) ([]atlasv2.AWSCloudProviderContainer, error)
+	AWSContainers(string) ([]*atlasv2.AWSCloudProviderContainer, error)
 	PeeringConnectionCreator
 }
 
 type GCPPeeringConnectionCreator interface {
-	GCPContainers(string) ([]atlasv2.GCPCloudProviderContainer, error)
+	GCPContainers(string) ([]*atlasv2.GCPCloudProviderContainer, error)
```

Comment:
This looks for me like non blocking openapi issue that makes usage of sdk look odd.

If someone wants optionality it should be on whole array. Having optional array items itself would be not likely to represent what api returns.

This PR is good and we done integration but I bet someone added optional on wrong object.

Reply:
That one is not due to openapi issues, but due to how oneOf models work. The method `GetActualInstance()` returns a pointer to the actual implementation. If we feel like this should not be an array of pointers, this can be done (this was done in the first commit)

Reply:
It doesn't matter then as there would be never nils there so we get same result.